aws/corehandlers/handlers.go

```diff
@@ -199,6 +199,10 @@ var ValidateResponseHandler = request.NamedHandler{Name: "core.ValidateResponseH
 // AfterRetryHandler performs final checks to determine if the request should
 // be retried and how long to delay.
 var AfterRetryHandler = request.NamedHandler{Name: "core.AfterRetryHandler", Fn: func(r *request.Request) {
+	if !aws.IsReaderSeekable(r.Body) {
+		r.Retryable = aws.Bool(false)
```

Comment:
This shouldn't get stomped over on line 207

Reply:
moved to the `Request.WillRetry` method to ensure it will always be the case.